fuzz/fuzz_targets/http_api.rs

```diff
@@ -0,0 +1,183 @@
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+use micro_http::Request;
+use once_cell::sync::Lazy;
+use std::os::unix::io::{AsRawFd, RawFd};
+use std::sync::mpsc::{channel, Receiver};
+use std::thread;
+use vmm::api::{http::*, ApiRequest, ApiResponsePayload};
+use vmm::{EpollContext, EpollDispatch};
+use vmm_sys_util::eventfd::EventFd;
+
+static ROUTES: Lazy<Vec<&Box<dyn EndpointHandler + Sync + Send>>> =
+    Lazy::new(|| HTTP_ROUTES.routes.values().collect());
+
+fuzz_target!(|bytes| {
+    if bytes.len() < 2 {
+        return;
+    }
+
+    let route = ROUTES[bytes[0] as usize % ROUTES.len()];
+    if let Some(request) = generate_request(&bytes[1..]) {
+        let mut epoll = EpollContext::new().unwrap();
+        let exit_evt = EventFd::new(libc::EFD_NONBLOCK).unwrap();
+        let api_evt = EventFd::new(libc::EFD_NONBLOCK).unwrap();
+        epoll.add_event(&exit_evt, EpollDispatch::Exit).unwrap();
+        epoll.add_event(&api_evt, EpollDispatch::Api).unwrap();
+
+        let (api_sender, api_receiver) = channel();
+
+        let http_api_thread = {
+            let exit_evt = exit_evt.try_clone().unwrap();
+            let api_evt = api_evt.try_clone().unwrap();
+            thread::Builder::new()
+                .name("http_api".to_string())
+                .spawn(move || {
+                    route.handle_request(&request, api_evt, api_sender);
+                    exit_evt.write(1).ok();
+                })
+                .unwrap()
+        };
+
+        http_receiver_stub(epoll.as_raw_fd(), api_evt, api_receiver);
+        http_api_thread.join().unwrap();
+    };
+});
+
+fn generate_request(bytes: &[u8]) -> Option<Request> {
+    let req_method = match bytes[0] % 3 {
+        0 => "GET",
+        1 => "PUT",
+        _ => "INVALID",
```

Comment:
You could also try other HTTP action types like `POST`

Reply:
For completeness, just added `PATCH` (supported by 'micro_http' crate and not being used in our code) and `POST` (not being supported by 'micro_http' crate).